File: app/models/approval.rb
=begin
    Approvals represent per-owner responses to requests to borrow
    Things.

    status defaults to "pending" at creation.
=end

class Approval < ActiveRecord::Base
  belongs_to :borrow_request
  belongs_to :user
  attr_accessible :status, :inclusion => { :in => %w(approved rejected pending),
    :message => "%{value} is not a valid status" }

  has_one :thing, :through => :borrow_request
  has_one :borrower, :through => :borrow_request, :source => :user
end

File: app/assets/javascripts/models/approval.js
Consolidator.Approval = DS.Model.extend({
    borrowRequest: DS.belongsTo("Consolidator.BorrowRequest"),
    user: DS.belongsTo("Consolidator.User"),

    status: DS.attr("string"),

    // Really hasOne relationships
    thing: DS.belongsTo("Consolidator.Thing"),
    borrower: DS.belongsTo("Consolidator.User")
});